.github/workflows/preliminary_checks.yml

```diff
@@ -93,6 +93,21 @@ jobs:
 
       - run: uv tool install clang-tidy==18.1.8
 
+      - name: Reinstall AWS SDK and s2n if setup scripts changed
```

Comment:
This is a temporary step since we made the AWS SDK version as required. Can we name this simply as `Reinstall AWS SDK` and add a comment above that this can be removed once this PR#14915 lands?